src/sentry/auth/system.py

```diff
@@ -0,0 +1,103 @@
+from __future__ import absolute_import, print_function
+
+from django.conf import settings
+from django.utils.crypto import constant_time_compare
+
+from sentry import options
+
+
+class SystemUser(object):
+    """
+    Singleton user representing a system that is part of Sentry.
+
+    The system user is a superuser that is not tied to an organization or real
+    user account. Unlike other superusers but similar to ``AnonymousUser``, it
+    has ``is_anonymous`` set.
+    """
+
+    id = -1  # Allow database queries
+
+    username = '<system>'
+    is_staff = True
+    is_active = True
+    is_superuser = True
+    is_system = True
+
+    def __str__(self):
+        return 'SystemUser'
+
+    def __eq__(self, other):
+        return isinstance(other, self.__class__)
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
+
+    def __hash__(self):
+        return 1  # singleton
+
+    def is_anonymous(self):
+        return True
+
+    def is_authenticated(self):
+        return True
+
+
+def is_internal_ip(request):
+    if settings.INTERNAL_IPS:
+        ip = request.META['REMOTE_ADDR']
+        if not any(ip in addr for addr in settings.INTERNAL_IPS):
+            return False
+
+    return True
+
+
+class SystemToken(object):
+    """
+    API token that gives superuser access to the system user.
+    """
+
+    id = '<system>'
+    token = '<system.secret-key>'
+    user = SystemUser()
+    application = None
+
+    @classmethod
+    def from_request(cls, request, token):
+        """Returns a system token if this is a valid system request."""
+        system_token = options.get('system.secret-key')
```

Comment:
Let's not re-use `system.secret-key`. I'd much rather we use another key explicitly set for this, or if you're feeling adventurous, a list of keys, so they could be rotated if needed, or to maintain multiple.

`system.secret-key` directly maps to Django's `SECRET_KEY` setting which is used for like, cryptographically signing cookies and whatnot. I'd rather other systems not have access to this value.

Reply:
Just discussed offline: This will become something similar to [`sentry:install-id`](https://github.com/getsentry/sentry/blob/54a792c30f6c75732eeec8ebdb2921bc4beda6c1/src/sentry/tasks/beacon.py#L41-L45)